azurelinuxagent/common/protocol/goal_state.py

```diff
@@ -128,40 +122,64 @@ def update_host_plugin_headers(wire_client):
         # Fetching the goal state updates the HostGAPlugin so simply trigger the request
         GoalState._fetch_goal_state(wire_client)
 
-    def update(self, force_update=False):
```

Comment:
The force_update flag is a leftover from a previous iteration. No longer needed.